Fabric.Terminology.Domain/Services/ValueSetService.cs

```diff
@@ -134,55 +135,78 @@ public bool NameIsUnique(string name)
             {
-                ValueSetId = valueSetId,
-                Name = name,
-                AuthoringSourceDescription = meta.AuthoringSourceDescription,
-                PurposeDescription = meta.PurposeDescription,
-                SourceDescription = meta.SourceDescription,
-                VersionDescription = meta.VersionDescription,
-                ValueSetCodes =
-                    valueSetCodeItems
-                        .Select(
-                            code => new ValueSetCode
-                            {
-                                Code = code.Code,
-                                CodeSystem = new ValueSetCodeSystem { Code = code.CodeSystemCode },
-                                Name = code.Name,
-                                RevisionDate = null,
-                                ValueSetId = valueSetId
-                            })
-                        .ToList()
-                        .AsReadOnly(),
-                IsCustom = true,
-                ValueSetCodesCount = valueSetCodeItems.Count()
+                ValueSetCodesCount = setCodes.Length,
+                IsCustom = true
             };
 
             Created?.Invoke(this, valueSet);
 
             return Attempt<IValueSet>.Successful(valueSet);
         }
 
-        // TODO need a table to insert/update
+        /// <summary>
+        /// Saves a <see cref="IValueSet"/>
+        /// </summary>
+        /// <param name="valueSet">The <see cref="IValueSet"/> to be saved</param>
+        /// <remarks>
+        /// At this point, we can only save "new" value sets.  Updates are out of scope at the moment - To be discussed.
+        /// </remarks>
         public void Save(IValueSet valueSet)
         {
-            throw new System.NotImplementedException();
+            if (valueSet.IsCustom && this.isCustomValue.Get(valueSet) && valueSet.IsNew())
+            {
+                Saving?.Invoke(this, valueSet);
+
+                var attempt = this.repository.Add(valueSet);
+                if (attempt.Success && attempt.Result.HasValue)
+                {
+                    Saved?.Invoke(this, attempt.Result.Single());
+                    return;
+                }
+
+                if (!attempt.Exception.HasValue)
+                {
+                    throw new ValueSetOperationException(
+                        "An exception was not returned by the attempt to save a ValueSet but the save failed.");
+                }
+
+                throw attempt.Exception.Single();
+            }
+
+            throw new InvalidOperationException("ValueSet was not a custom value set and cannot be saved.");
         }
 
-        // TODO need a table to delete
         public void Delete(IValueSet valueSet)
         {
-            // assert is custom
-            throw new System.NotImplementedException();
+            if (valueSet.IsCustom && this.isCustomValue.Get(valueSet) && !valueSet.IsNew())
```

Comment:
This was brought up in the meeting with the Terminology team and this will be refactored once the `ClientTermFLG` field is added to the `Terminology.ValueSetDescription` view.